wire-ios/Wire-iOS/Sources/URLActionRouter.swift

```diff
@@ -147,9 +148,10 @@ extension URLActionRouter: PresentationDelegate {
         switch action {
         case .connectBot:
             presentConfirmationAlert(title: UrlAction.title, message: UrlAction.ConnectToBot.message, decisionHandler: decisionHandler)
-        case .accessBackend(configurationURL: let configurationURL):
-            guard SecurityFlags.customBackend.isEnabled else { return }
-            presentCustomBackendAlert(with: configurationURL)
+        case .accessBackend(let url):
+            // Switching backend is handled below, so pass false here.
+            decisionHandler(false)
+            switchBackend(configURL: url)
```

Comment:
Ideally we should refactor this so that the router isn't responsible for processing the deep link. I tried this but it's a rabbit hole of refactoring so I left as it is for now.

Reply:
Should we create a ticket to follow up?

Reply:
Yep I will.